charts/actions-runner-controller/values.yaml

```diff
@@ -39,6 +39,20 @@ image:
   actionsRunnerRepositoryAndTag: "summerwind/actions-runner:latest"
   dindSidecarRepositoryAndTag: "docker:dind"
   pullPolicy: IfNotPresent
+  # The default image-pull secret name for self-hosted runner container.
+  # It's added to spec.ImagePullSecrets of self-hosted runner pods.
+  # actionsRunnerImagePullSecret: someSecretName
+
+
+imagePullSecrets:
+- name: github-actions-docker-config
+
+# Optionally create a secret with the dockerconfigjson used for the runner pod spec
+# imagePullSecretsDockerConfig:
+#   secretName: someSecretName
+#   registry: somePrivateRegistry
+#   username: someUsername
+#   password: somePassword
```

Comment:
Can we make this out of the scope of this PR or this chart, or can we make it more generalized like `serviceAccount`?

I mean, there are might be many image pull secrets you'd want to manage as part of your actions-runner-controller instance. Then you'd want ability to  define one or more image pull secrets like this:

```suggestion
# Optionally create a secret with the dockerconfigjson used for the runner pod spec
imagePullSecrets:
- name: blah
  type: kubernetes.io/dockerconfigjson
  stringDataTemplate: |
    .dockerconfigjson: |-
      {
      "auths": {
        "{{ .Values.imagePullSecretsDockerConfig.registry }}": {
            "username": "{{ .Values.imagePullSecretsDockerConfig.username }}",
            "password": "{{ .Values.imagePullSecretsDockerConfig.password }}"
          }
      }
    }
```

In the manager secret template we'd do this with the `tpl` function so that you can freely hard-code all the JSON values within the values.yaml or you can render the dockerconfigjson with values loaded via e.g. helm-secrets.

```
{{ range $_, $c := .Values.imagePullSecrets }}
---
apiVersion: v1
kind: Secret
type: {{ $c.type }}
metadata:
  name: {{ $c.name }}
stringData:
  {{ tpl $c.stringDataTemplate $ }}
{{ end }}
```

Reply:
Thanks for the feedback and review @mumoshu. I've improved it on 2d62544e1a043834affc9bac91ff621924a2f013 and made it a bit more generic [here](https://github.com/actions-runner-controller/actions-runner-controller/pull/921/commits/2d62544e1a043834affc9bac91ff621924a2f013#diff-3dfd28f4387ada0d16b7e7cdc2efd38b3724c82f6821b4f0fa5b3045ff9788c4R36-R53). Let me know what your thoughts are 😃